pandas/core/ops.py

```diff
@@ -299,7 +300,16 @@ def __init__(self, left, right, name, na_op):
         self.is_datetime64tz_rhs = is_datetime64tz_dtype(rvalues)
         self.is_datetime_rhs = self.is_datetime64_rhs or self.is_datetime64tz_rhs
         self.is_timedelta_rhs = is_timedelta64_dtype(rvalues)
+        if self.rhs_type_from_series is None:
```

Comment:
why can't you do the check that you have below 

```
elif isinstance(values, pd.Series) and (
 +                    is_timedelta64_dtype(values) or is_datetime64_dtype(values)):
```

here rather than carry this instance variable around?